.dockerignore

```diff
@@ -1,88 +1,75 @@
-# Git
-.git
-.gitignore
-
-# Byte-compiled / optimized / DLL files
-__pycache__/
-*/__pycache__/
-*/*/__pycache__/
-*/*/*/__pycache__/
-*.py[cod]
-*/*.py[cod]
-*/*/*.py[cod]
-*/*/*/*.py[cod]
-
-# C extensions
-*.so
-
-# Distribution / packaging
-.Python
-env/
-build/
-develop-eggs/
-dist/
-downloads/
-eggs/
-lib/
-lib64/
-parts/
-sdist/
-var/
-*.egg-info/
-.installed.cfg
-*.egg
-
-# PyInstaller
-#  Usually these files are written by a python script from a template
-#  before PyInstaller builds the exe, so as to inject date/other infos into it.
-*.manifest
-*.spec
-
-# Installer logs
-pip-log.txt
-pip-delete-this-directory.txt
-
-# Unit test / coverage reports
-htmlcov/
-.tox/
-.coverage
-.cache
-nosetests.xml
-coverage.xml
-
-# Translations
-*.mo
-*.pot
-
-# Django stuff:
-*.log
-
-# Sphinx documentation
-docs/_build/
-
-# PyBuilder
-target/
-
-# Virtual environment
-.env/
-.venv/
-venv/
-
-# PyCharm
-.idea
-
-# Python mode for VIM
-.ropeproject
-*/.ropeproject
-*/*/.ropeproject
-*/*/*/.ropeproject
-
-# Vim swap files
-*.swp
-*/*.swp
-*/*/*.swp
-*/*/*/*.swp
-
-# Yaml files
-*.yml
-*.yaml
+**/*.egg
+**/*.egg-info/
+**/*.log
+**/*.manifest
+**/*.mo
+**/*.pot
+**/*.py[cod]
+**/*.so
+**/*.spec
+**/*.swp
+**/*.yaml
+**/*.yml
+**/*/*.py[cod]
+**/*/*.swp
+**/*/*/*.py[cod]
+**/*/*/*.swp
+**/*/*/*/*.py[cod]
+**/*/*/*/*.swp
+**/*/*/*/.ropeproject
+**/*/*/*/__pycache__/
+**/*/*/.ropeproject
+**/*/*/__pycache__/
+**/*/.ropeproject
+**/*/__pycache__/
+**/.AppleDouble
+**/.DS_Store
+**/.LSOverride
+**/.Python
+**/.cache
+**/.circleci
+**/.coverage
+**/.dockerignore
+**/.env
+**/.env.*
+**/.env/
+**/.git
+**/.gitignore
+**/.hokusai-tmp
+**/.idea
+**/.installed.cfg
+**/.pytest_cache
+**/.python-version
+**/.ropeproject
+**/.secrets.baseline
+**/.tox/
+**/.venv/
+**/Dockerfile
+**/LICENSE
+**/Makefile
+**/README.md
+**/VERSION
+**/__pycache__/
+**/build/
+**/coverage.xml
+**/develop-eggs/
+**/dist/
+**/docs/_build/
+**/downloads/
+**/eggs/
+**/env
+**/env/
+**/hokusai
+**/htmlcov/
+**/lib/
+**/lib64/
+**/log
+**/nosetests.xml
+**/parts/
+**/pip-delete-this-directory.txt
+**/pip-log.txt
+**/sdist/
+**/target/
+**/tmp
+**/var/
+**/venv/
```

Comment:
The old style of grouping patterns by type was too tedious to work with, so I just removed all comments, and sorted the list. Adding `**` in front so that the patterns are matched at all directory levels. This is a difference between `.dockerignore` and `.gitignore`, if I understand it right. By default, `.gitignore` matches at all directory levels, whereas `.dockerignore` only matches at repo root. Please double check me on that, and if that's correct, we should perhaps update [templates](https://github.com/artsy/artsy-hokusai-templates/blob/main/rails-puma/.dockerignore.j2).

Reply:
Not official docs but [this](https://zzz.buzz/2018/05/23/differences-of-rules-between-gitignore-and-dockerignore/) confirms the path understanding

Reply:
Thanks @ovasdi.